src/controllers/post.ts

```diff
@@ -7,34 +7,100 @@ import {
 	remove,
 	allPostsByUser,
 	feed,
+	like,
+	unlike,
+	getComments,
 } from '../lib/post';
+import { IGetUserAuthRequest } from '../types/Request';
 
-const getAllPosts = async (req: Request, res: Response) => {
-	const response = await getAll(req, res);
+const getAllPosts = async (req: IGetUserAuthRequest, res: Response) => {
+	const { page = 1, limit = 10 } = req.query;
+	const { user } = req;
+	const response = await getAll(page as number, limit as number, user.userRole);
+	return res.status(response.status).json(response);
 };
 
 const getPost = async (req: Request, res: Response) => {
-	const response = await getById(req, res);
+	const { id } = req.params;
+	const response = await getById(id);
+	return res.status(response.status).json(response);
 };
 
-const createPost = async (req: Request, res: Response) => {
-	const response = await create(req, res);
+const createPost = async (req: IGetUserAuthRequest, res: Response) => {
+	const { title, content } = req.body;
+	const { user } = req;
+	if (req.user.userRole === 'MODERATOR') {
+		return res.status(403).json({
+			message: 'Moderators cannot create posts',
+		});
+	}
+	const response = await create(title, content, user._id);
+	return res.status(response.status).json(response);
 };
 
-const updatePost = async (req: Request, res: Response) => {
-	const response = await update(req, res);
+const updatePost = async (req: IGetUserAuthRequest, res: Response) => {
+	const { id } = req.params;
+	const { title, content } = req.body;
+	const { user } = req;
+	const response = await update(id, title, content, user._id);
+	return res.status(response.status).json(response);
 };
 
-const deletePost = async (req: Request, res: Response) => {
-	const response = await remove(req, res);
+const deletePost = async (req: IGetUserAuthRequest, res: Response) => {
+	const { id } = req.params;
+	const { user } = req;
+	const response = await remove(id, user._id, user.userRole);
+	return res.status(response.status).json(response);
 };
 
-const getAllPostsByUser = async (req: Request, res: Response) => {
-	const response = await allPostsByUser(req, res);
+const getAllPostsByUser = async (req: IGetUserAuthRequest, res: Response) => {
+	const { page = 1, limit = 10, desc = true } = req.query;
+	const { user } = req;
+	const response = await allPostsByUser(
+		page as number,
+		limit as number,
+		desc as string,
+		user._id
+	);
+	return res.status(response.status).json(response);
 };
 
-const getFeed = async (req: Request, res: Response) => {
-	const response = await feed(req, res);
+const getFeed = async (req: IGetUserAuthRequest, res: Response) => {
+	const { page = 1, limit = 10, desc = true } = req.query;
+	const { user } = req;
+	if (!user.paid) {
```

Comment:
this can read better if we go like 
if (user.paid) {

}
else
{

}

Reply:
Ok. Will do that in the future.